firebase/authentication.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  browserLocalPersistence,
+  GoogleAuthProvider,
+  setPersistence,
+  signOut,
+  signInWithPopup,
+} from "firebase/auth";
+import { auth } from "./firebase";
+import type { User } from "firebase/auth";
+
+export function getCurrentUser(): User | null {
+  return auth.currentUser;
+}
+
+auth.onAuthStateChanged(() => {
+  const user = getCurrentUser();
+  if (user) {
+    console.log("User is logged in");
```

Comment:
Again, Vercel doesn't need this information, but we do

Reply:
Has been removed